### PR TITLE
Pass message headers to Tinode.publish explicitly.

### PR DESCRIPTION
**Pull request by aforge**
Needed in order to correctly save message headers for outgoing messages.

### CODE REVIEW

File: TinodeSDK/Tinode.swift [hunk 1]
```diff
@@ -1060,18 +1060,28 @@ public class Tinode {
         return sendWithPromise(payload: msg, with: msgId)
     }
 
-    internal func publish(topic: String, head: [String:JSONValue]?, content: Drafty) -> PromisedReply<ServerMessage>? {
+    public static func draftyHeaders(for message: Drafty) -> [String: JSONValue] {
+        var head = [String: JSONValue]()
```
Comment (or-else) [inline TinodeSDK/Tinode.swift]: Swift 4 and 5 canonical is 
`var head: [String:JSONValue] = []`

File: TinodeSDK/Tinode.swift [hunk 2]
```diff
@@ -1060,18 +1060,28 @@ public class Tinode {
         return sendWithPromise(payload: msg, with: msgId)
     }
 
-    internal func publish(topic: String, head: [String:JSONValue]?, content: Drafty) -> PromisedReply<ServerMessage>? {
+    public static func draftyHeaders(for message: Drafty) -> [String: JSONValue] {
```
Comment (or-else) [inline TinodeSDK/Tinode.swift]: Just a question. Should we check `message.isPlain` and skip creating headers if true?

File: TinodeSDK/Tinode.swift [hunk 3]
```diff
@@ -1060,18 +1060,28 @@ public class Tinode {
...
+    public func publish(topic: String, head: [String:JSONValue]?, content: Drafty) -> PromisedReply<ServerMessage>? {
         let msgId = getNextMsgId()
         let msg = ClientMessage<Int, Int>(
             pub: MsgClientPub(id: msgId, topic: topic, noecho: true, head: head, content: content))
         return sendWithPromise(payload: msg, with: msgId)
     }
-
+    /*
     public func publish(topic: String, data: Drafty) -> PromisedReply<ServerMessage>? {
         return publish(topic: topic,
```
Comment (or-else) [inline TinodeSDK/Tinode.swift]: Delete?

File: TinodeSDK/Tinode.swift (hunk 2 above)
Comment (aforge) [reply TinodeSDK/Tinode.swift]: Sure. Originally, that's how the code was written.
I made it so in order to be consistent with Android.

File: TinodeSDK/Tinode.swift (hunk 1 above)
Comment (aforge) [reply TinodeSDK/Tinode.swift]: Done.

File: TinodeSDK/Tinode.swift (hunk 3 above)
Comment (aforge) [reply TinodeSDK/Tinode.swift]: This should be gone. A stale snapshot.

**Pull request merged**